uaclient/tests/test_cli_attach.py

```diff
@@ -62,6 +97,26 @@ def test_token_is_a_required_argument(self, _m_getuid):
             action_attach(args, FakeConfig())
         assert status.MESSAGE_ATTACH_REQUIRES_TOKEN == str(e.value)
 
+    @mock.patch(M_PATH + "contract.request_updated_contract")
+    def test_status_updated_when_auto_enable_fails(
```

Comment:
This test passes against master as-is, so I don't think it's testing the added code appropriately.

Reply:
done. added a missing mock to this test which only succeeded as we reached out to the real contract server due to contract.get_available_resources()

Reply:
This test still hasn't been updated. If I apply this patch to your branch:

```diff
diff --git a/uaclient/cli.py b/uaclient/cli.py
index 3f44c0e..3e02312 100644
--- a/uaclient/cli.py
+++ b/uaclient/cli.py
@@ -329,11 +329,9 @@ def _attach_with_token(
         with util.disable_log_to_console():
             logging.exception(exc)
         print(ua_status.MESSAGE_ATTACH_FAILURE)
-        cfg.status()  # Persist updated status in the event of partial attach
         return 1
     except exceptions.UserFacingError as exc:
         logging.warning(exc.msg)
-        cfg.status()  # Persist updated status in the event of partial attach
         return 1
     contract_name = cfg.machine_token["machineTokenInfo"]["contractInfo"][
         "name"
```

then this test still passes:

```
$ pytest

<snip>

==================== 553 passed in 1.87 seconds =====================
```

which demonstrates that it is not testing the change proposed in this PR. Can you modify it so that it is testing the new behaviour, please?

Reply:
Yep, good point, I wasn't asserting deep enough. Added logic to exercise poth cfg.status() paths upon UrlError and UserFacingError.